lib/internal/test_runner/test.js

```diff
@@ -288,6 +288,13 @@ class Test extends AsyncResource {
     };
     this.waitingOn = 0;
     this.finished = false;
+
+    if (!testOnlyFlag && only) {
```

Comment:
what about `test.runOnly`?

Reply:
I've pushed an update so that the test will emit a diagnostic for the following scenario. 
```js
// running with `--test-only`
const { test } = require('node:test');

test('top level test', async (t) => {
  t.runOnly(true);
  await t.test('this subtest is run');
});

/*
TAP version 13
# Subtest: top level test
    # Subtest: this subtest is run
    ok 1 - this subtest is run
      ---
      duration_ms: 4.329238
      ...
    # 'only' does not work when node is started without the --test-only command-line option
    1..1
ok 1 - top level test
  ---
  duration_ms: 5.897932
*/

```